exporter/metrics.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/brigadecore/brigade/sdk/v2"
+	"github.com/brigadecore/brigade/sdk/v2/authn"
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/brigadecore/brigade/sdk/v2/meta"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	totalRunningJobs = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "brigade_running_jobs_total",
+		Help: "The total number of running jobs",
+	})
+
+	totalPendingJobs = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "brigade_pending_jobs_total",
+		Help: "The total number of pending jobs",
+	})
+
+	allWorkersByPhase = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "brigade_all_workers_by_phase",
+		Help: "All workers separated by phase",
+	}, []string{"workerPhase"})
+
+	allRunningWorkersDuration = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "brigade_all_running_workers_duration",
+		Help: "The duration of all running workers",
+	}, []string{"worker"})
+
+	totalUsers = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "brigade_users_total",
+		Help: "The total number of users",
+	})
+
+	totalServiceAccounts = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "brigade_service_accounts_total",
+		Help: "The total number of service accounts",
+	})
+
+	totalProjects = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "brigade_projects_total",
+		Help: "The total number of brigade projects",
+	})
+)
+
+func recordMetricsLoop(
+	ctx context.Context,
+	client sdk.APIClient,
+	scrapeInterval time.Duration,
+) {
+	ticker := time.NewTicker(scrapeInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			recordMetrics(client)
+		case <-ctx.Done():
+			return
+		}
+	}
```

Comment:
On each run through the loop, we wait until we get a tick from the ticker or the context is canceled (i.e. the loop is interrupted).